libmamba/src/api/install.cpp

```diff
@@ -933,7 +933,15 @@ namespace mamba
                     std::vector<std::string> f_specs;
                     for (auto& line : file_contents)
                     {
-                        if (line[0] != '#' && line[0] != '@')
+                        auto lstrip_line = util::lstrip(line);
+
+                        // Skip comment lines and empty lines
+                        // Comment lines start with '#' or '@' preceded by whitespaces or tabs
+                        auto is_comment = util::starts_with(lstrip_line, "#")
+                                          || util::starts_with(lstrip_line, "@");
+                        auto is_empty = lstrip_line.empty();
+
+                        if (!is_comment && !is_empty)
                         {
                             f_specs.push_back(line);
```

Comment:
I think it's the stripped version that should be pushed?

Reply:
If we want to push the stripped version (a `std::stringview` in this context), we need to reallocate a string, which can be costly. I preferred not to for now.

What do you prefer?

Reply:
I would just re-assign the stripped version in the same string, we dont need the unstripped version and we can (maybe) reuse the memory from the initial string?
Not a big deal anyway.